chat/tasks.py

```diff
@@ -15,7 +15,18 @@ def create_chat_for_puzzle(puzzle_id):
         msg = f"{puzzle.name} has been unlocked!"
         puzzle.chat_room.send_and_announce_message_with_embedded_urls(msg, puzzle)
     except Exception as e:
-        logger.warn(f"Chat operations failed with error: {e}")
+        logger.warn(f"create_chat_for_puzzle failed with error: {e}")
+
+
+@shared_task
+def cleanup_puzzle_channels(puzzle_id):
+    puzzle = Puzzle.objects.get(id=puzzle_id)
+    # check that puzzle wasn't unsolved between when task was queued and now
```

Comment:
It's pretty edge case-y, but if a puzzle was solved, unsolved, and re-solved within 30m, this could delete the channels sooner than expected. Should we maybe add an explicit solved timestamp to the Puzzle model? Looks like it might make the stats slightly more robust too: https://github.com/cardinalitypuzzles/cardboard/blob/195dea2eb57eddcd1bfc9da3b02d8788a8922844/hunts/chart_utils.py#L43

Reply:
Could https://github.com/cardinalitypuzzles/cardboard/blob/master/puzzles/models.py#L164 work for both? I guess it's not really more robust than what's in chart_utils.py right now.

Reply:
Yeah sounds good, fine w/ me if you want to just use that for now. Introducing a separate solve timestamp has its own downsides since now we have to keep it accurate.